src/cmd/rmi.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright © 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/containers/toolbox/pkg/podman"
+	"github.com/containers/toolbox/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	rmiFlags struct {
+		deleteAll   bool
+		forceDelete bool
+	}
+)
+
+var rmiCmd = &cobra.Command{
+	Use:   "rmi",
+	Short: "Remove one or more toolbox images",
+	Run: func(cmd *cobra.Command, args []string) {
+		rmi(args)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(rmiCmd)
+
+	flags := rmiCmd.Flags()
+	flags.BoolVarP(&rmiFlags.deleteAll, "all", "a", false, "Remove all toolbox images")
+	flags.BoolVarP(&rmiFlags.forceDelete, "force", "f", false, "Force the removal of used toolbox images")
+}
+
+func rmi(args []string) {
+	if rmiFlags.deleteAll {
+		logrus.Info("Fetching images with label=com.github.debarshiray.toolbox=true")
+		args := []string{"--filter", "label=com.github.debarshiray.toolbox=true"}
+		Dimages, err := podman.GetImages(args...)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		logrus.Info("Fetching images with label=com.github.containers.toolbox=true")
+		args = []string{"--filter", "label=com.github.containers.toolbox=true"}
+		Cimages, err := podman.GetImages(args...)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		var idKey string
+		if podman.CheckVersion("1.8.2") < 0 {
+			idKey = "ID"
+		} else {
+			idKey = "id"
+		}
+		images := utils.JoinJSON(idKey, Dimages, Cimages)
+		for _, image := range images {
+			logrus.Infof("Deleting image %s", image[idKey].(string))
```

Comment:
Isn't `image[idKey]` already a `string`? Is the cast necessary?

Reply:
Children of `image` are of type `interface{}` which is a vague type that can store anything -> no generics. So the cast is necessary (though I did not cast it in some cases). In some cases casting is not needed (for `logrus` or `fmt`) but I would call it safer.